### PR TITLE
Update `tower` and `tower-util` and prep for release

### PR DESCRIPTION
**Pull request by LucioFranco**

### CODE REVIEW

**Comment by LucioFranco** [comment]
This depends on #376 and #377

File: tower/tests/builder.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#![cfg(feature = "broken")]
```
Comment (davidbarsky) [inline tower/tests/builder.rs]: I don't see this one documented or used anywhere else. What's up?

Comment (LucioFranco) [reply tower/tests/builder.rs]: Yeah, I need buffer to get merged before I can renable this

Comment (LucioFranco) [reply tower/tests/builder.rs]: I meant to open this as a draft.

Comment (davidbarsky) [reply tower/tests/builder.rs]: Gotcha.

Comment (davidbarsky) [reply tower/tests/builder.rs]: If you rebase this atop of the buffer changes, I'll be able to approve?

**Pull request merged**